scripts/lib/source.py

```diff
@@ -131,6 +136,19 @@ def categorize_files(files):
         result.classify('objc', 'related file', filename)
         continue
 
+      if _in_directories(filename, OBJC_DIRS):
+        result.classify('objc', 'directory', filename)
+        continue
+
+      raise Exception(textwrap.dedent(
```

Comment:
`Exception` is usually just a base class to be extended by the other errors. You really want an InvalidStateError, which doesn't exist. The closest that I can see is ValueError. RuntimeError might also be ok. More options here: https://docs.python.org/3/library/exceptions.html

Reply:
I went for `NotImplementedError` to more strongly convey that this is a work in progress and can be changed.